modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java

```diff
@@ -334,4 +318,22 @@ public boolean isDynamic() {
         .filter(ob -> ob instanceof ConnectionProviderObjectBuilder)
         .map(ob -> ((ConnectionProviderObjectBuilder) ob).providerModel);
   }
+
+  /**
+   * Used to preserve exception throwing behaviour in {@link #getConfiguration} as the method {@link #createConfiguration} can no longer throw
+   * checked exception as it is used inside {@link Map#computeIfAbsent}.
+   */
+  private class WrappingRuntimeException extends RuntimeException {
+
+    private final Exception wrappedException;
```

Comment:
You can use the cause instead of creating the instance variable.

Reply:
but getCause() returns a Throwable type instead of an Exception type. I would either need to cast it or change the getConfiguration signature to "throws Throwable".

Reply:
👍